tests/load_tests/sys_profiling.py

```diff
@@ -16,12 +16,34 @@ def get_size_gb(bytes):
     return bytes / (1024**3)
 
 
+def get_worker_memory(process_keyword):
+    top = 0
+    acc = 0
+    count = 0
+    for proc in psutil.process_iter(['name', 'cmdline']):
+        try:
+            cmdline = ' '.join(proc.cmdline()) if proc.cmdline() else ''
+            if process_keyword in cmdline:
+                top = max(top, proc.memory_info().rss)
+                acc += proc.memory_info().rss
+                count += 1
+        except (psutil.NoSuchProcess, psutil.AccessDenied,
+                psutil.ZombieProcess):
+            pass
+    avg = acc / count if count > 0 else 0
+    return top, avg
+
+
 def monitor_system():
     # Initialize peak tracking variables
     interval = 0.2
     peak_cpu = 0
     peak_memory_percent = 0
     peak_memory_used = 0
+    nonblocking_worker_memory_peak = 0
+    blocking_worker_memory_peak = 0
+    nonblocking_worker_memory_avg_peak = 0
+    blocking_worker_memory_avg_peak = 0
```

Comment:
Should we rename these to `long` and `short` as well?

Reply:
good catch